File: 2021/2021W03/PreppindataW03.py
import pandas as pd

xlsx = pd.ExcelFile(r'F:\Data\PD 2021 Wk 3 Input.xlsx')
temp = []
for sheet in [sh for sh in xlsx.sheet_names]:
    df = pd.read_excel(xlsx,sheet)
    df.insert(0,'Store',sheet)
    temp.append(df)
df = pd.concat(temp) 

df = pd.melt(df, id_vars=['Date', 'Store'], var_name='VAR', value_name='Products Sold')
temp = df['VAR'].str.split('\s-\s', n = 1, expand = True) 
df['Customer Types'] = temp[0] 
df['Product'] = temp[1]
df['Quarter'] = df['Date'].dt.quarter

finalA = df.groupby(['Product','Quarter'], as_index=False).agg({'Products Sold':'sum'})
finalB = df.groupby(['Store','Customer Types','Product'], as_index=False).agg({'Products Sold':'sum'})

File: 2021/2021W03/preppindataW03.R
library(readxl)
library(dplyr)
library(tidyr)
library(purrr)
library(lubridate)

input <- "F:/Data/PD 2021 Wk 3 Input.xlsx"
df <-  input %>%
  excel_sheets() %>% 
  set_names() %>%
  map_df(
    ~ read_excel(path = input, sheet = .x,),
    .id = 'Store') %>%
  pivot_longer(cols=-c('Date','Store'), 
               names_to=c('Customer Types','Product'), names_pattern='(\\w+)\\W+(\\w+)',
               values_to = 'Products Sold') %>%
  mutate('Quarter' = quarter(`Date`))

finalA <- df %>%
  group_by(`Product`,`Quarter`) %>%
  summarise('Products Sold' = sum(`Products Sold`))

finalB <- df %>%
  group_by(`Store`,`Customer Types`,`Product`) %>%
  summarise('Products Sold' = sum(`Products Sold`))

View(finalA)
View(finalB)
